lib/asset-rev.js

```diff
@@ -24,9 +25,19 @@ function AssetRev(inputTree, options) {
   this.ignore = options.ignore;
   this.description = options.description;
 
+  // first pass - excludes replaceable source code
+  this.exclude = exclude.slice();
+  this.replaceExtensions.forEach(function(ext) { this.exclude.push('**/*.' + ext); }, this);
```

Comment:
`this.exclude` is a list of strings used by minimatch for glob matching, as well as a special-cased direct substring search (indexOf).

    Fingerprint.prototype.excludeContains = function (relativePath) {
      for (var i = 0; i < this.exclude.length; i++) {
        // glob expression
        if (this.exclude[i].match(relativePath)) {
          return true;
        }
        // allow 'fonts' and 'assets/fonts' to match 'assets/font/foo.ttf'
        if (relativePath.indexOf(this.exclude[i].pattern) !== -1) {
          return true;
        }
      }
      return false;
    }

First pass takes advantage of this by excluding the "replaceable source code" file extensions using glob syntax: `js` => `**/*.js`.

Which will match all files ending in the excluded file extension.


I'd also prefer to remove the substring matching altogether, as it can have unexpected behavior as illustrated by your example. But this would be a breaking change...